packages/core/src/dispatcher.ts

```diff
@@ -156,6 +156,27 @@ export class Dispatcher {
     })
   }
 
+  async getIpfsBlock(cid: CID): Promise<Uint8Array> {
+    return await this._shutdownSignal.abortable((signal) => {
+      return this._ipfs.block.get(cid, { signal })
+    })
+  }
+
+  /**
+   * Stores all the blocks in the given CAR file into the local IPFS node.  Uses the blocks API
+   * instead of the dag.import API because the dag.import API requires the CAR file to be a fully
+   * self-contained graph but our CAR file may have references to IPLD objects that aren't included
+   * in the CAR file itself.
+   * @param car
+   */
+  async storeCarFile(car: CAR): Promise<void> {
+    return await this._shutdownSignal.abortable(async (signal) => {
+      for (const block of car.blocks) {
+        await this._ipfs.block.put(block.payload, { signal })
```

Comment:
Why don't we just use ipfs.dag.import so that we only need to make one http call?

Reply:
IPFS `ipfs.dag.import` assumes that a CAR file contains a _full subgraph_. This is not the case for us: we do not store commits previous to anchor commits.

Reply:
"Elsewhere" means when creating a commit, I guess. When we store a commit, (a) we have the full commit data, (b) we have previous commits available on the same node. So, it is a different thing.

Reply:
> Just because we have all the previous commits on the node doesn't mean we have the entire DAG because anchors link to multiple streams technically

Heck, you are right.

> What happens if you try to import a car file without a full DAG?

IPFS would hang till it downloads the DAG.

Reply:
Or maybe we need to patch IPFS?

Reply:
Looks like it's an option?
https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/DAG.md#options-3

Reply:
Looks like you are already setting this to false

Reply:
Although, @ukstv do you know if I set that `pinRoots` flag to false, would I be able to use `dag.import` here safely?  It'd be nicer than having to send each block one at a time

Reply:
Ok, great! 🙏 🙏 

So should this PR change it's implementation to use `dag.import`?

Reply:
It should.
@stbrody